include_core/omrhookable.h

```diff
@@ -65,12 +70,56 @@ typedef struct J9HookInterface {
 #define J9HOOK_AGENTID_DEFAULT  ((uintptr_t)1)
 #define J9HOOK_AGENTID_LAST  ((uintptr_t)-1)
 
+/* time threshold (=100 milliseconds) for triggering the tracepoint  */
+#define OMRHOOK_DEFAULT_THRESHOLD_IN_MILLISECONDS_WARNING_CALLBACK_ELAPSED_TIME	100
+
+/*
+ * array of OMREventInfo4Dump is added in individual hookInterface by Hook generation tool to avoid
+ * rumtime native memory allocation(malloc), this macro would pass the array pointer and size from
+ * the individual hookInterface to J9CommonHookInterface, then hookable code and javadump code can
+ * easily access OMREventInfo4Dump[].
+ */
+#define OMRHOOK_INITIALIZE_HOOKDUMP(interface) \
+	do { \
+		interface.common.eventSize = sizeof(interface.flags) / sizeof(interface.flags[0]); \
+		interface.common.infos4Dump = &interface.infos4Dump[0]; \
+	} while (0)
+
+/* this macro would initialize hookInterface and dump struct for hook */
+#define J9HOOK_INITIALIZE_INTERFACE_AND_DUMP(hookInterface, portLib) ({ \
+	intptr_t result = 0; \
+	do { \
+		J9HookInitializeInterface(J9_HOOK_INTERFACE(hookInterface), portLib, sizeof(hookInterface)); \
+		if (0 == result) { \
```

Comment:
If you look at the macros for HOOK_FLAGS and HOOK_RECORD could you do something similar so we do not have to have the extra infos4Dump pointer and so we do not need this code.